elide-core/src/main/java/com/yahoo/elide/core/FilterScope.java

```diff
@@ -27,6 +29,23 @@ public FilterScope(RequestScope requestScope, Class<?> resourceClass) {
         this.permissions = requestScope.getDictionary().getPermissionsForClass(resourceClass, ReadPermission.class);
     }
 
+    /*
+     *This constructor with FilterExpression allows merging the filterExpression parsed from URL and the one from
+     *FilterExpressionCheck.
+     */
+    public FilterScope(RequestScope requestScope, Class<?> resourceClass, FilterExpression permissionExpression) {
+        this.requestScope = requestScope;
+        this.permissions = requestScope.getDictionary().getPermissionsForClass(resourceClass, ReadPermission.class);
+        if (permissionExpression != null) {
+            if (this.requestScope.getGlobalFilterExpression() != null) {
+                this.requestScope.setGlobalFilterExpression(
+                        new AndFilterExpression(this.requestScope.getGlobalFilterExpression(), permissionExpression));
```

Comment:
This logic might make more sense in the getGlobalFilterExpression function in RequestScope.